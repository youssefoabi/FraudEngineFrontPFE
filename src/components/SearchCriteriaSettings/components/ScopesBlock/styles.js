export default function style() {
  return {
    ButtonWrapper: {
      marginTop: 8,
      alignItems: 'center',
    },
  };
}
