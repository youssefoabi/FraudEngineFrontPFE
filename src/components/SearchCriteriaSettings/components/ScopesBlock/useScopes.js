import { filter, keys, map, pathOr } from 'ramda';
import { isNilOrEmpty } from 'ramda-adjunct';
import { useCallback, useMemo } from 'react';

import useScopes from '../../../../screens/ScopesManagement/useScopes';

const useScopesList = ({ handleChange, engine }) => {
  const { scopes: allScopes } = useScopes();
  const { scopes: engineScopes, selectedScope } = engine || {};

  const handleSelectScope = ({ value }) => {
    const scope = pathOr({}, ['value'], value);
    const { name } = scope;
    handleChange({ id: 'selectedScope', value: { ...scope, label: name } });
  };

  const addScope = () => {
    if (isNilOrEmpty(selectedScope)) return;

    const scopeId = pathOr(null, ['scopeId'], selectedScope);
    handleChange({
      id: 'scopes',
      value: {
        ...(engineScopes || {}),
        [scopeId]: { scopeId, name: pathOr('', ['name'], selectedScope), order: keys(engineScopes).length },
      },
    });
  };

  const mapScope = useCallback(
    scope => {
      if (isNilOrEmpty(scope)) return null;

      const { name, id } = scope;
      const findScope = pathOr(null, [id], engineScopes);
      const isNewScope = findScope === null;
      if (!isNewScope) return null;

      return { value: { ...scope, scopeId: id }, label: name };
    },
    [engineScopes],
  );

  const scopesSelect = useMemo(() => {
    const mappedScopes = map(mapScope, allScopes);

    return filter(scope => scope !== null, mappedScopes);
  }, [allScopes, mapScope]);

  return {
    scopes: scopesSelect,
    handleSelectScope,
    addScope,
  };
};

export default useScopesList;
