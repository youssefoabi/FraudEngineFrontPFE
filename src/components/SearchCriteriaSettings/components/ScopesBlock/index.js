import { Button, DataEdit, GridItem, useStyles } from '@cdiscount/backoffice-ui';
import { func, shape } from 'prop-types';
import Add from '@material-ui/icons/Add';
import Grid from '@material-ui/core/Grid';
import React from 'react';

import ScopesList from './components/ScopesList';
import styles from './styles';
import useScopes from './useScopes';

const ScopesBlock = ({ engine, handleChange }) => {
  const { handleSelectScope, scopes: allScopes, addScope } = useScopes({
    handleChange,
    engine,
  });
  const classes = useStyles(styles);

  return (
    <>
      <GridItem title="Choix des gammes">
        <Grid container spacing={3}>
          <Grid item xs={6} md={4} lg={4}>
            <DataEdit
              fields={[
                {
                  id: 'selectedScope',
                  type: 'select',
                  label: 'Gamme 1',
                  selectProps: {
                    options: allScopes,
                  },
                },
              ]}
              values={engine}
              onChange={handleSelectScope}
            />
          </Grid>
          <GridItem xs={6} md={3} lg={3}>
            <Button
              className={classes.ButtonWrapper}
              size="large"
              icon={Add}
              color="tertiary"
              variant="contained"
              label="AJOUTER UNE GAMME"
              onClick={addScope}
            />
          </GridItem>
        </Grid>
        <ScopesList engine={engine} handleChange={handleChange} />
      </GridItem>
    </>
  );
};

ScopesBlock.propTypes = {
  engine: shape({}),
  handleChange: func,
};

ScopesBlock.defaultProps = {
  engine: null,
  handleChange: Function.Prototype,
};

export default ScopesBlock;
