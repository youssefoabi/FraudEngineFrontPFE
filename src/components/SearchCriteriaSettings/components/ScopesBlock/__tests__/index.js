import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import ScopesBlock from '../';

jest.mock('react-beautiful-dnd', () => ({
  Droppable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  Draggable: ({ children }) =>
    children(
      {
        draggableProps: {
          style: {},
        },
        innerRef: jest.fn(),
      },
      {},
    ),
  DragDropContext: ({ children }) => children,
}));

jest.mock('react-redux', () => ({
  useDispatch: () => ({}),
  useSelector: () => ({}),
}));

describe(ScopesBlock.name, () => {
  itRendersCorrectly(<ScopesBlock />);
});
