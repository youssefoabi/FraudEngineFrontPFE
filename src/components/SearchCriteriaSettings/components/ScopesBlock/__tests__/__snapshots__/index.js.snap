// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScopesBlock renders correctly 1`] = `
<Fragment>
  <h
    fullheight={false}
    title="Choix des gammes"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={3}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        lg={4}
        md={4}
        xs={6}
      >
        <f
          config={null}
          fields={
            Array [
              Object {
                "id": "selectedScope",
                "label": "Gamme 1",
                "selectProps": Object {
                  "options": Object {},
                },
                "type": "select",
              },
            ]
          }
          inputs={null}
          onChange={[Function]}
          onInit={[Function]}
          title={null}
          values={null}
        />
      </WithStyles(ForwardRef(Grid))>
      <h
        fullheight={false}
        lg={3}
        md={3}
        title={null}
        xs={6}
      >
        <y
          className="makeStyles-ButtonWrapper-1"
          color="tertiary"
          icon={
            Object {
              "$$typeof": Symbol(react.memo),
              "compare": null,
              "displayName": "AddIcon",
              "muiName": "SvgIcon",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            }
          }
          label="AJOUTER UNE GAMME"
          onClick={[Function]}
          size="large"
          variant="contained"
        />
      </h>
    </WithStyles(ForwardRef(Grid))>
    <ScopesList
      engine={null}
    />
  </h>
</Fragment>
`;
